### PR TITLE
Free hosting trial: Remove A/B test

### PR DESCRIPTION
**Pull request by zaguiini**
Related to pdKhl6-2T3-p2#comment-4921.

## Proposed Changes

This PR removes the A/B test while making the free hosting trial available for all folks browsing `/setup/new-hosted-site` that are eligible for a hosting trial (i.e., new site and/or never had a trial before.)

We’ve already had a couple of use cases where it would’ve been beneficial to have it enabled to 100% (agency pitches, Reddit ads), and the cash sales and support impact are not drastically different, so it's not harmful.

## Testing Instructions

In a fresh account, navigate to `/setup/new-hosted-site` and verify that the "Try for free" CTA shows up in the plans step. 

Start the free trial and verify that browsing `/setup/new-hosted-site` again hides the "Try for free" CTA.

### CODE REVIEW

File: client/landing/stepper/declarative-flow/new-hosted-site-flow.ts [hunk 1]
```diff
@@ -149,9 +147,6 @@ const hosting: Flow = {
 			() => {
 				if ( currentStepSlug === undefined ) {
 					resetOnboardStore();
-					if ( query.get( 'campaign' ) === 'reddit' ) {
-						reduxDispatch( savePreference( 'hosting-trial-campaign', 'reddit' ) );
-					}
```
Comment (zaguiini) [inline client/landing/stepper/declarative-flow/new-hosted-site-flow.ts:154]: The Reddit campaign work overrides the A/B test to enable the free trial. Now that we’ll stop the A/B test, it doesn’t make sense to keep the Reddit campaign logic

**Review by vykes-mac — APPROVED** [review]
Changes looks good and it works as expected.

**Review by p-jackson — APPROVED** [review]
I've re-based and re-tested. Everything is working as expected. I can see in `localStorage` that no A/B test assignments are being made, and I can still use `?flags=-plans/hosting-trial` to make the trial UI go away.

It makes sense to keep the feature flag around for now so we can quickly disable it over the holidays if necessary.

I'm leaving the DO NOT MERGE label in place because of this thread: p1702423372363699-slack-C0347E545HR

**Comment by p-jackson** [comment]
We're deploying p1702569185629659/1702423372.363699-slack-C0347E545HR

**Pull request merged**